### PR TITLE
streamlink 0.4.0 (new formula)

### PR DESCRIPTION
**Pull request by rakkesh**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/streamlink.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+    system "python", *Language::Python.setup_install_args(libexec)
+
+    bin.install Dir[libexec/"bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/streamlink --version 2>&1")
```
Comment (bfontaine) [inline Formula/streamlink.rb]: Can’t we write a test that does something more substantial than e.g. `--help` or `--version`?

Comment (rakkesh) [inline Formula/streamlink.rb]: A more functional test would depend on an external resource, such as, a video stream or a youtube video. The current `version` test should be good enough to check for all dependent libraries.

Comment (neutric) [inline Formula/streamlink.rb]: YouTube video you say? Check out the [test in `youtube-dl.rb`](https://github.com/Homebrew/homebrew-core/blob/ef3bec8fe8a39616942f04cf664fa92aeb5b3cb3/Formula/youtube-dl.rb#L24-L25) that does just that.

Comment (rakkesh) [inline Formula/streamlink.rb]: I don't mind adding an external video source for the tests, I was considering if it ever gets removed and the test will fail.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by fxcoudert** [comment]
Thanks @rakkesh for your contribution to Homebrew! We couldn't do it without people like you

**Pull request closed without merging**